ports/rp2/mphalport.c

```diff
@@ -176,11 +176,22 @@ void mp_hal_delay_ms(mp_uint_t ms) {
     }
 }
 
-uint64_t mp_hal_time_ns(void) {
+static uint64_t time_ns_offset;
```

Comment:
Rename to something like `boot_time_offset_from_epoch_in_us`.  (the problem with the original name is the "time" in "time_us_64" is "since boot", whereas the "time" in "mp_hal_time_ns" is "since epoch").

Reply:
Renamed to `time_us_64_offset_from_epoch`.